flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/FieldsDataType.java

```diff
@@ -57,24 +63,105 @@ public FieldsDataType(LogicalType logicalType, List<DataType> fieldDataTypes) {
     }
 
     @Override
-    public DataType notNull() {
+    public FieldsDataType notNull() {
```

Comment:
this is API breaking, we should not do this. as we have seen recently on the dev@ ML